Source/Core/Core/IOS/ES/Views.cpp

```diff
@@ -135,16 +135,56 @@ ReturnCode ESDevice::GetV0TicketFromView(const u8* ticket_view, u8* ticket) cons
   return IPC_SUCCESS;
 }
 
+ReturnCode ESDevice::GetV1TicketFromView(const u8* ticket_view, u8* ticket,
```

Comment:
The `optional` size seems unnecessary.  Couldn't you just use `u32*`.  Then just pass `nullptr` in the case below where `ticket != nullptr`.

Reply:
On top of what @iwubcode said _(regarding `optional` and `nullptr`)_, is there a reason not to move the V1 specific code to `GetV0TicketFromView` and maybe rename the function since V0/V1 almost look like the same?